src/Chapter3.hs

```diff
@@ -480,6 +546,81 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city in total.
 -}
 
+data Person 
+    = Person 
+    deriving (Show)
+
+data House = House 
+    { houseResidents :: [Person]
+    } deriving (Show)
+
+data CityBuilding 
+    = Church 
+    | Library 
+    deriving (Show)
+
+data Castle = Castle
+    { castleName    :: String,
+      cityWalls     :: Int 
```

Comment:
Nice data type!
I think that walls could be represented by a more suitable data type, as currently it allows to have negative number of walls 🙂